### ISSUE TITLE
Feature request to enable CodeLens to appear above decorators (TypeScript)

### ISSUE THREAD

**Issue by JohnWeisz** [Microsoft/vscode#22698]
This is how CodeLens currently looks when it appears on a property with decorators applied (same happens with methods and classes):

![codelens-decorators](https://cloud.githubusercontent.com/assets/13468659/23973361/15d6e5f6-09d6-11e7-939b-84fc8cf2c2ad.png)

(TypeScript)

It would be nice to either have CodeLens annotations above decorators, or have a setting to enable this, as it's a bit hard to follow the code right now. Moving the decorator to the same line where the property/method declaration is is certainly better for now, but it's not always feasible.

**Comment by mjbvz** [comment]
The current code lens placement is consistent with Visual Studio's code lens placement. It also always ensures that the code lens is in a consistent location relative to the referenced symbol
